core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -1050,8 +1053,16 @@ impl WorkflowMachines {
                 }
                 WFCommand::AddActivity(attrs) => {
                     let seq = attrs.seq;
+                    let use_compat = self.determine_use_compatible_flag(
```

Comment:
Would have been cleaner if the workflow machines context (with the task queue) propagated to the internal state machines and more of the logic moved in there.

Reply:
I agree but I want to deal with that more generally. I want to do a change where we stop cloning a bunch of these strings everywhere and instead pass some refcounted propery bag type things. I made an item https://github.com/temporalio/sdk-core/issues/567